Seminars/Sem.04/Pract.04/Task02.cpp

```diff
@@ -0,0 +1,224 @@
+			index++;
+		}
+		this->condition[index] = '\0';
+	}
+	void setPoints(int points) {
+		if (points >= 0) {
+			this->points = points;
+		}
+	}
+	void changeName(const char name[]) {
+		setName(name);
+	}
+	void changeCondition(const char condition[]) {
+		setCondition(condition);
+
+	}
+	void changePoints(int points) {
+		setPoints(points);
+	}
+};
+class Exam {
+private:
+	Task tasks[NUMBER_OF_TASKS];
+	int numOfTasks;
+	int minPoints;
+
+public:
+	Exam(const Task tasks[], int minPoints, int numOfTasks) {
+		setTasks(tasks, numOfTasks);
+		setMinPoints(minPoints);
+
+	}
+
+	void setTasks(const Task tasks[], int numOfTasks){
+
+		if (numOfTasks >= 0) {
+			this ->numOfTasks = numOfTasks;
+		}
+		
+		for (int i = 0; i < numOfTasks; i++) {
+			this->tasks[i] = tasks[i];
+		}
+
+     }
+	void setMinPoints(int minPoints) {
+		if (minPoints >= 0) {
+			this->minPoints = minPoints;
+		}
+	}
+	void changeMinPoints(int minPoints) {
+		setMinPoints(minPoints);
+	}
+	int getMinPoints()const {
+		return minPoints;
+	}
+	
+	int getMaxPoints()const {
+		int maxPoints = 0;
+		for (int i = 0; i < numOfTasks; i++) {
+			maxPoints += tasks[i].getPoints();
+		}
+		return maxPoints;
+
+	}
+
+
+	void writeToFile(const char* filename)const {
+		std::ofstream file(filename,std:: ios::out |std:: ios::binary);
+
+		if (!file.is_open()) {
+			std::cerr << "ERROR OPENING THE "<<filename<< "FILE";
+			return;
+		}
+		file.write((const char*)&minPoints, sizeof(int));
+		file.write((const char*)&numOfTasks, sizeof(int));
+		
+		for (int i = 0; i < numOfTasks; i++) {
+			
+			const char* name = tasks[i].getName();
+			size_t LengthName = strlen(tasks[i].getName());
+			const char* condition = tasks[i].getCondition();
+			size_t ConditionLength = strlen(tasks[i].getCondition());
+			const int points = tasks[i].getPoints();
+			file.write((const char*)&LengthName, sizeof(LengthName));
+			file.write(name, LengthName);
+			file.write((const char*)&ConditionLength, sizeof(ConditionLength));
+			file.write(condition, ConditionLength);
+			file.write((const char*)&points, sizeof(int));
+		}
+		file.close();
+
+
+	}
+	void readFromFile(std::ifstream& file) {
+	
+		int points;
+		file.read((char*)&points, sizeof(int));
+		int minPoints;
+		file.read((char*)&minPoints, sizeof(int));
+		setMinPoints(minPoints);
+		Task *tasks=new Task[numOfTasks];
+		for (int i = 0; i < numOfTasks; i++) {
+			size_t LengthName, ConditionLength;
+
+			file.read((char*)&LengthName, sizeof(LengthName));
+			char* name = new char[LengthName + 1];
+			file.read(name, LengthName);
+			name[LengthName] = '\0';
+
+			file.read((char*)&ConditionLength, sizeof(ConditionLength));
+			char* condition = new char[ConditionLength + 1];
+			file.read(condition, ConditionLength);
+			condition[ConditionLength] = '\0';
+
+			int points;
+			file.read((char*)&points, sizeof(int));
+			tasks[i] = Task(name, condition, points);
+			delete[] name;
+			delete[] condition;
+			setTasks(tasks, i);
```

Comment:
The idea here is wrong, I presume that setTasks should be out of the body  this for loop